kazoo/protocol/connection.py

```diff
@@ -808,23 +806,11 @@ def _authenticate_with_sasl(self, host, timeout):
             try:
                 response = sasl_cli.process(challenge=challenge)
             except puresasl.SASLError as err:
-                six.reraise(
-                    SASLException,
-                    SASLException("library error: %s" % err),
-                    sys.exc_info()[2],
-                )
-            except puresasl.SASLProtocolException as err:
-                six.reraise(
-                    AuthFailedError,
-                    AuthFailedError("protocol error: %s" % err),
-                    sys.exc_info()[2],
-                )
-            except Exception as err:
-                six.reraise(
-                    AuthFailedError,
-                    AuthFailedError("Unknown error: %s" % err),
-                    sys.exc_info()[2],
-                )
+                raise SASLException("library error") from err
+            except puresasl.SASLProtocolException as exc:
+                raise AuthFailedError("protocol error") from exc
+            except Exception as exc:
+                raise AuthFailedError("Unknown error") from exc
```

Comment:
I know this was already partially discussed, but why mix `err` and `exc`? Here it just adds confusion... pick one in this section for consistency...